src/main/daml/Daml/Finance/Interface/Claims/Util/Lifecycle.daml

```diff
@@ -74,6 +75,23 @@ lifecycleClaims observableCids acquisitionTime claims events = do
   observe <- collectObservables observableCids
   fmap mconcat . unzip <$> mapA (applyEvents observe acquisitionTime events) claims
 
+-- | Lifecycle a set of claims at specified events.
```

Comment:
I would isolate the netting logic into a function with signature `net : [Pending] -> [Pending]`. You can then compose it with lifecycle
```
remaining , pending <- lifecycle
let nettedPending = net pending
```

I would be descriptive in the description of the function to make it clear according to which equality function you are netting (in this specific case, you are netting regardless of tags, whereas in other scenarios we might want to net on (tag, instrument))

Reply:
Good point, I think this looks better now.